src/libpriv/rpmostree-origin.cxx

```diff
@@ -102,6 +102,14 @@ sync_treefile (RpmOstreeOrigin *self)
   g_assert_no_error (local_error);
 }
 
+static void
+sync_origin (RpmOstreeOrigin *self)
+{
+  g_autoptr (GKeyFile) kf = rpmostree_origin_dup_keyfile (self);
```

Comment:
I think I would find this less confusing if we called `CXX_MUST_VAL (rpmostreecxx::treefile_to_origin (**origin->treefile));` instead (they're the same thing I know, but it would make it extra clear we actually are creating a new keyfile here)